src/SuperMint.sol

```diff
@@ -24,10 +24,13 @@ contract SuperMint is IERC721Receiver {
     uint256,
     bytes memory
   ) public virtual override returns (bytes4) {
-    if (!complete) {
-      complete = true;
-      implementation.mint(maxMints - 1);
+    /**@notice - check if maxMints is not 0 and then reduce its value
+    before reentrance. When it's value drops below 1, reentrancy stops.
+    */
+    if (maxMints - 1 != 0) {
+      maxMints--;
+      implementation.mint(iterator);
     }
```

Comment:
`maxMints - 1` and `maxMints--` are doing the same thing here, imo we should just have:
```
if (maxMints-- != 0) {
  implementation.mint(iterator);
}
```